### PR TITLE
include layer opacity info for embedded layers

### PR DESCRIPTION
**Pull request by mind84**
Backport for #4964

Funded by Faunalia

### CODE REVIEW

**Review by Gustry — APPROVED** [review]
LGTM according to LGTM on the parent PR ;-)

**Comment by Gustry** [comment]
I wonder how CI could pass, I'm confused as well, from other PR on the 3.8 branch.

But now, when I'm on the landing page, I have a 500 error from PHP : 

```
[0]	Call to a member function xpath() on null
  /srv/lzm/lizmap/modules/lizmap/lib/Project/QgisProject.php	654
 /
 array (   'module' => 'view',   'action' => 'default:index', )
 ref:Unknown referer

	0	Lizmap\Project\QgisProject->xpathQuery()	/srv/lzm/lizmap/modules/lizmap/lib/Project/QgisProject.php : 390
	1	Lizmap\Project\QgisProject->setLayerOpacity()	/srv/lzm/lizmap/modules/lizmap/lib/Project/QgisProject.php : 346
	2	Lizmap\Project\QgisProject->setPropertiesAfterRead()	/srv/lzm/lizmap/modules/lizmap/lib/Project/Project.php : 236
	3	Lizmap\Project\Project->readProject()	/srv/lzm/lizmap/modules/lizmap/lib/Project/Project.php : 148
	4	Lizmap\Project\Project->__construct()	/srv/lzm/lizmap/modules/lizmap/lib/Project/Repository.php : 294
	5	Lizmap\Project\Repository->getProject()	/srv/lzm/lizmap/modules/lizmap/lib/Project/Repository.php : 388
	6	Lizmap\Project\Repository->getProjectsMetadata()	/srv/lzm/lizmap/modules/lizmap/classes/lizmapRepository.class.php : 166
	7	lizmapRepository->getProjectsMetadata()	/srv/lzm/lizmap/modules/view/zones/main_view.zone.php : 64
	8	main_viewZone->_prepareTpl()	/srv/lzm/lizmap/vendor/jelix/jelix/lib/jelix/utils/jZone.class.php : 282
	9	jZone->_createContent()	/srv/lzm/lizmap/vendor/jelix/jelix/lib/jelix/utils/jZone.class.php : 217
	10	jZone->getContent()	/srv/lzm/lizmap/vendor/jelix/jelix/lib/jelix/utils/jZone.class.php : 373
	11	jZone::_callZone()	/srv/lzm/lizmap/vendor/jelix/jelix/lib/jelix/utils/jZone.class.php : 114
	12	jZone::get()	/srv/lzm/lizmap/vendor/jelix/jelix/lib/jelix/tpl/jTpl.class.php : 52
	13	jTpl->assignZone()	/srv/lzm/lizmap/modules/view/controllers/default.classic.php : 158
	14	defaultCtrl->index()	/srv/lzm/lizmap/vendor/jelix/jelix/lib/jelix/core/jCoordinator.class.php : 266
	15	jCoordinator->process()	/srv/lzm/lizmap/www/index.php : 23
```

Which looks linked to this PR ? 

Am I wrong ?

**Comment by mind84** [comment]
Hi @Gustry,

I'm surprised too! 

Anyway, the issue is related to the `xpathQuery` function, it's different between versions:

versions >= **3.9**:
https://github.com/3liz/lizmap-web-client/blob/0766319f723776c0e7528d5bfee4aa568d5c45ba/lizmap/modules/lizmap/lib/Project/QgisProject.php#L656

version **3.8**:

https://github.com/3liz/lizmap-web-client/blob/d691b68c19cbaa6a3fe63f82ca6a3f395ede02ca/lizmap/modules/lizmap/lib/Project/QgisProject.php#L654

Basically in 3.8 the property `$xml` is accessed directly while in newer versions Lizmap first checks if this property exists.

This is the commit:

https://github.com/3liz/lizmap-web-client/commit/1e9be742fe5b70ff744ff40ad66951460ccf0a8d

Maybe coluld be backported to 3.8? Any drawbaks? @rldhont,

**Comment by rldhont** [comment]
@mind84 the commit https://github.com/3liz/lizmap-web-client/commit/1e9be742fe5b70ff744ff40ad66951460ccf0a8d is part of https://github.com/3liz/lizmap-web-client/pull/4765 and was made has a first step for https://github.com/3liz/lizmap-web-client/pull/4819

If you can provide a fix to release_3_8, it's welcome.
